add Coder hosted getting started guide (#571)
cdr/docs: guides/saas/getting-started.md

---
title: Getting started with Coder's hosted beta
description: Get started with a hosted Coder deployment.
---

To help you get started, we've put together this guide to help you understand
how the Coder hosted beta works and how to access your deployment.

> ## What is Coder hosted beta?

Reviewer: @CoderErik created a lovely graphic demonstrating how this works. thoughts on adding a version of this in a later revision?

We also discussed adding a second graphic to demonstrate how the clients actually have a point-to-point connection with workspaces. Thoughts on that as well?

@jmilling33 could probably work his magic on it

[Shared Hosting.pdf](https://github.com/cdr/docs/files/7078351/Shared.Hosting.pdf)

Reviewer: whoops here is the updated graphic

[Shared Hosting (1).pdf](https://github.com/cdr/docs/files/7078391/Shared.Hosting.1.pdf)

Author: Definitely think @CoderErik's graphic would fit well in here.

Reviewer 2: Here's a link to png version
https://lucid.app/publicSegments/view/7e950228-6b5a-47eb-b9d4-ed5a82e5898d/image.png

Author: @jmilling33 from a design standpoint, should we defer to you on that? @CoderErik's graphic is accurate, but I just want to make sure we keeps things consistent.

Reviewer 3: I am happy to restyle the graphic to make sure it's consistent with our branding. When do you need it by?

Reviewer 2: @jmilling33 Just shared the original file with you. Check email for the link.

Reviewer: @jmilling33 no hard deadline. It would be nice to update this doc in ~1 week or so via another PR.

For now, we can leave the illustration out

The Coder hosted beta is a hybrid cloud offering of Coder's self-hosted
platform. With this model, Coder hosts the control plane, and you host your
compute and development workspaces. The benefits include:

- Automatic upgrades
- No installation or DNS configuration
- Managed logging & monitoring

> Curious how this works? Here's a breakdown of the architecture:

Reviewer: I suggest adding a CTA for folks to try it, as we do on out home page

```suggestion
If you interested in trying the limited beta, [contact us](https://coder.com/contact?note=I%20would%20like%20to%20try%20the%20hosted%20offer.%0A%0ANumber%20of%20developers%3A%0A%0AUse%20case%3A))

Curious how this works? Here's a breakdown of the architecture:
```

Reviewer: JFYI: this probably isn't linted correctly. i did from the web ui

Author: Good call - will add.

**Coder's infrastructure**:

- The `coderd` service - Responsible for rendering the dashboard UI,
  provisioning workspaces and user authentication, including a variety of other functions
  - Your `coderd` instance is _not_ shared across other Coder accounts. Each deployment
      receives their own `coderd` instance
- A PostgreSQL DB - Stores metadata related to your Coder instance,
  such as user information, session tokens, etc.
  - Coder uses a single PostgreSQL instance with each deployment having their
    own database within the instance

**Your infrastructure**:

- Kubernetes cluster - Hosts your Coder workspaces, in addition to your source
  code, which is stored on [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)
  that are mounted to each workspace pod

`coderd` connects to your infrastructure through a service account created in
your cluster, enabling you to create workspaces in your cluster.

## Accessing your Coder hosted beta

> 1. Navigate to your deployment URL and login with the provided credentials:

Reviewer: These details are currently emailed to the admin. Perhaps we should mention that here in case they forget what their URL or password is?

Author: You're right - in the current state, it assumes the reader knows they've been emailed the credentials. I can clarify.

cluster, enabling you to create workspaces.

## Accessing your Coder hosted beta

1. Navigate to your deployment URL (e.g., `<your-name>.coder.com`).

1. Log in with the email address you provided to Coder the password Coder
   provided you.

1. Once you've logged in, you'll be prompted to change your temporary password.

You're in! At this point, you'll need to

> [connect your Kubernetes cluster to Coder](#link-here) before you can create

Reviewer 4: gotta remember to update this placeholder

Author: before we merge - I think we need to update this to link to #576.

---
title: Getting started with Coder's hosted beta
description: Get started with a hosted Coder deployment.
---

To help you get started, we've put together this guide to help you understand
how the Coder hosted beta works and how to access your deployment.

## What is Coder hosted beta?

The Coder hosted beta is a hybrid cloud offering of Coder's self-hosted
platform. With this model, Coder hosts the control plane, and you host your
compute and development workspaces. The benefits include:

- Automatic upgrades

> - No installation or DNS configuration

Reviewer 5: ```suggestion
- DNS or TLS configuration
```

They still have to `install` a provider so I'm not sure "no installatino" is valid.

- Managed logging & monitoring

If you interested in trying the limited beta, [contact us](https://coder.com/contact?note=I%20would%20like%20to%20try%20the%20hosted%20offer.%0A%0ANumber%20of%20developers%3A%0A%0AUse%20case%3A))

Curious how this works? Here's a breakdown of the architecture:

**Coder's infrastructure**:

- The `coderd` service - Responsible for rendering the dashboard UI,
  provisioning workspaces and user authentication, including a variety of other functions
  - Your `coderd` instance is _not_ shared across other Coder accounts. Each deployment
      receives their own `coderd` instance
- A PostgreSQL DB - Stores metadata related to your Coder instance,
  such as user information, session tokens, etc.

>   - Coder uses a single PostgreSQL instance with each deployment having their
>     own database within the instance

Reviewer 5: ```suggestion
```

Unnecessary and subject to change as we scale.

**Your infrastructure**:

- Kubernetes cluster - Hosts your Coder workspaces, in addition to your source
  code, which is stored on [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)
  that are mounted to each workspace pod

`coderd` connects to your infrastructure through a service account created in
your cluster, enabling you to create workspaces in your cluster.

## Accessing your Coder hosted beta

1. Navigate to your deployment URL and login with the following credentials:

   - Deployment URL should be `<your-name>.coder.com`

>    - Username should be `admin`

Reviewer 5: ```suggestion
   - Username should be the email address you provided
```

   - Password should be similar to `Lv7...k3`

> > These credentials are emailed to your Coder admin.
> 
> 1. Once logged in, you'll be prompted to change your temporary password
> 
> 1. Enter the license file provided to you by email

Reviewer 5: ```suggestion
You will receive the credentials from Coder when the instance is provisioned.
When you log in, you'll be prompted to change your temporary password
```

We set up the license for them.
